### PR TITLE
Remove the unused AEADBCryptHandles.cs

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
Introduced in #52030, `AEADBCryptHandles.cs` is almost identical to another new file of that PR, `BCryptAEADHandleCache.cs`. Contrary to the latter however, the former is not used at all (searching `AEADBCryptHandles.cs` gave only one results on the file itself; presumably it was left by accident and slipped code review). This PR removes it.

### CODE REVIEW

**Review by stephentoub — APPROVED** [review]
LGTM if CI is green.  Thanks.

**Review by bartonjs — APPROVED** [review]
Surprise had me waiting until CI confirmed it first, but, yeah.  "So long as CI agrees..."

**Comment by bartonjs** [comment]
Weird.  Thanks for helping keep us tidy :smile:.

**Pull request merged**
